File: 274. H-Index/274. H-Index.java
class Solution {

  public int hIndex(int[] citations) {
    int n = citations.length;
    int[] numCitations = new int[n + 1];

    for (int c : citations) {
      numCitations[Math.min(c, n)]++;
    }

    int papers = 0;
    for (int h = n; h > -1; h--) {
      papers += numCitations[h];
      if (papers >= h) {
        return h;
      }
    }

    return 0;
  }
}

File: 274. H-Index/274. H-Index.py
from typing import List


class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        num_citations = [0] * (n + 1)

        for c in citations:
            num_citations[min(c, n)] += 1

        papers = 0
        for h in range(n, -1, -1):
            papers += num_citations[h]
            if papers >= h:
                return h
